webviz_subsurface/containers/_summary_stats.py

```diff
@@ -59,62 +91,90 @@ def layout(self):
-                                     'statistics': False}]),
-                (get_summary_data, [{'ensemble_path': self.ensemble_path,
-                                     'sampling': self.sampling,
-                                     'statistics': True}])]
+        return [(get_summary_data, [{'ensemble_paths': self.ensemble_paths,
+                                     'column_keys': self.column_keys,
+                                     'sampling': self.sampling}]),
+                (get_summary_stats, [{'ensemble_paths': self.ensemble_paths,
+                                      'column_keys': self.column_keys,
+                                      'sampling': self.sampling}])]
 
 
 @cache.memoize(timeout=cache.TIMEOUT)
 @webvizstore
-def get_summary_data(ensemble_path, sampling,
-                     statistics=False) -> pd.DataFrame:
+def get_summary_data(ensemble_paths, sampling, column_keys) -> pd.DataFrame:
+    """ Loops over given ensemble paths, extracts smry-data and concates them
+    into one big df. An additional column ENS gets added for eacht ens-path
+    to seperate the ensambles.
 
-    ens = scratch_ensemble('', ensemble_path)
-    if statistics:
-        return ens.get_smry_stats(time_index=sampling)
-    else:
-        return ens.get_smry(time_index=sampling)
+    Dash functions take positional args., so order matters. """
+
+    ens_data_dfs = []
+
+    for ensemble_path in ensemble_paths:
+        ensemble_df = scratch_ensemble('', ensemble_path).get_smry(
+            time_index=sampling, column_keys=column_keys)
+        ensemble_df['ENS'] = ensemble_path.replace(
+            '/scratch/troll_fmu/', '')
+        ens_data_dfs.append(ensemble_df)
+
+    return pd.concat(ens_data_dfs)
 
 
 @cache.memoize(timeout=cache.TIMEOUT)
-def render_realization_plot(ensemble_path, sampling, vector):
+@webvizstore
+def get_summary_stats(ensemble_paths, column_keys, sampling) -> pd.DataFrame:
+    """ Loops over given ensemble paths, extracts smry-data and concates them
+    into one big df. An additional column ENS gets added for eacht ens-path
+    to seperate the ensambles.
+
+    Dash functions take positional args., so order matters. """
 
-    data = get_summary_data(ensemble_path,
-                            sampling)[['REAL', 'DATE', vector]]
+    df_ens_set = []
+
+    for path in ensemble_paths:
+        stats = scratch_ensemble('', path).get_smry_stats(
+                time_index=sampling, column_keys=column_keys)
+        stats['ENS'] = path.replace(
+                       '/scratch/troll_fmu/', '')
+        df_ens_set.append(stats)
+
+    return pd.concat(df_ens_set)
+
+
+@cache.memoize(timeout=cache.TIMEOUT)
+def render_realization_plot(ensemble_paths, sampling, column_keys, vector):
+    """ returns a single dcc.Graph """
+
+    summary_stats = get_summary_data(ensemble_paths, column_keys, sampling
+                                     )[['REAL', 'DATE', 'ENS', vector]]
 
     traces = [{
```

Comment:
There's a bug here that plots missing realizations as a trace.
Test using e.g. : /scratch/fmu/sago/3_r001_reek/realization-*/iter-4

![img](https://user-images.githubusercontent.com/16436291/56659090-8db6a780-669c-11e9-8109-db0e0b183c56.png)